src/zepben/evolve/streaming/get/network_consumer.py

```diff
@@ -130,98 +130,109 @@ async def y():
 
     async def _get_network_hierarchy(self) -> GrpcResult[NetworkHierarchy]:
         return await self.try_rpc(self._handle_network_hierarchy)
 
     async def _get_feeder(self, service: NetworkService, mrid: str) -> GrpcResult[MultiObjectResult]:
-        feeder_response = await self._get_identified_object(service, mrid)
-        if feeder_response.was_failure:
-            return feeder_response
+        feeder_response = (await self._get_identified_object(service, mrid)).throw_on_error()
         feeder: Feeder = feeder_response.result
 
         if not feeder:
             return GrpcResult(result=None)
 
-        equipment_objects = await self._get_identified_objects(service, service.get_unresolved_reference_mrids(resolver.ec_equipment(feeder)))
-        if equipment_objects.was_failure:
-            return equipment_objects
-
-        resolvers = list()
-        resolvers.append(resolver.normal_energizing_substation(feeder))
-        for equip in feeder.equipment:
-            try:
-                for terminal in equip.terminals:
-                    resolvers.append(resolver.connectivity_node(terminal))
-                if isinstance(equip, Conductor):
-                    resolvers.append(resolver.per_length_sequence_impedance(equip))
-                    resolvers.append(resolver.asset_info(equip))
-            except AttributeError:
-                pass  # Not ConductingEquipment.
-            resolvers.append(resolver.psr_location(equip))
-
-        mrids = service.get_unresolved_reference_mrids(resolvers)
-        objects = await self._get_identified_objects(service, mrids)
-        if objects.was_failure:
-            return objects
-        objects.result.value.update(equipment_objects.result.value)  # Combine with previous results
-        objects.result.value[feeder.mrid] = feeder  # Add feeder to result
-        return GrpcResult(result=MultiObjectResult(objects.result.value, objects.result.failed.union(equipment_objects.result.failed)))
+        if not isinstance(feeder, Feeder):
+            return GrpcResult(result=ValueError(f"Requested mrid {mrid} was not a Feeder, was {type(feeder)}"))
+
+        mor = MultiObjectResult()
+        res = await self._get_objects(service, service.get_unresolved_reference_mrids_by_resolver(resolver.ec_equipment(feeder)))
+        mor.value.update(res.value)
+        # We need to resolve all the unresolved references for each piece of equipment, but then we need to
+        # also do the same for the resolutions, and so on so forth until the entire tree of references originating from the Feeder has been resolved,
+        # however we don't know how long this tree is until we've resolved everything. So we start by resolving all the equipment, then we iteratively
+        # descend the tree by checking for unresolved references for the objects we just resolved in the previous iteration.
+        to_resolve = self._get_unresolved_mrids(service, res.value.keys())
+        while True:
```

Comment:
So this needs to be reworked as discussed. We should resolve references unless they represent some form of connectivity that will blow out to resolving everything. Initial items to restrict:
* Feeders should not automatically resolve equipment
* ConnectivityNodes should not automatically resolve its terminals.

Reply:
@gcarwardine thinking more on this, another way that this can creep is via `UsagePoint` and `Meter` or via `OperationalRestriction`. There may be more but resolving all relationships may not be the best way forward, although the creep in these casses will be much lower as the connectivity will not be traversed

Reply:
yeah i had a think about this and came to the conclusion that if we exclude the ConnectivityNode -> Terminal relationship when resolving we shouldn't creep beyond what's actually included in the OperationalRestriction/UsagePoint/etc, which you'd probably want anyway if you were pulling in the feeder?